zerver/signals.py

```diff
@@ -72,8 +73,11 @@ def email_on_new_login(sender: Any, user: UserProfile, request: Any, **kwargs: A
 
         context = common_context(user)
         context['user_email'] = user.email
-        context['login_time'] = timezone_now().strftime('%A, %B %d, %Y at %I:%M%p ') + \
-            timezone_get_current_timezone_name()
+        user_tz = user.timezone
+        if user_tz == '':
+            user_tz = timezone_get_current_timezone_name()
+        local_time = timezone_now().astimezone(get_timezone(user_tz))
+        context['login_time'] = local_time.strftime('%A, %B %d, %Y at %I:%M%p ') + user_tz
```

Comment:
Can you update the tests in `SendLoginEmailTest` to verify this logic?  I think probably making `test_send_login_emails_if_send_login_email_setting_is_true` set a timezone at the beginning, mock `time.time()`, and do some checks on the email content makes the most sense.

Reply:
@timabbott , I've updated `test_send_login_emails_if_send_login_email_setting_is_true` as you suggested. Thanks!